lib/acts_as_xapian/acts_as_xapian.rb

```diff
@@ -117,13 +117,21 @@ def self.prepare_environment
     config_file = Rails.root.join("config","xapian.yml")
     @@config = File.exist?(config_file) ? YAML.load_file(config_file)[environment] : {}
 
-    # figure out where the DBs should go
-    if config['base_db_path']
-      db_parent_path = Rails.root.join(config['base_db_path'])
-    else
-      db_parent_path = File.join(File.dirname(__FILE__), 'xapiandbs')
+    if config['base_db_path'] && config['absolute_base_db_path']
+      msg = 'Only set one of base_db_path OR absolute_base_db_path'
+      raise ArgumentError, msg
     end
 
+    # figure out where the DBs should go
+    db_parent_path =
+      if config['base_db_path']
+        Rails.root.join(config['base_db_path'])
+      elsif config['absolute_base_db_path']
+        config['absolute_base_db_path']
+      else
```

Comment:
```suggestion
    # figure out where the DBs should go
    db_parent_path =
      if config['base_db_path']
        config['base_db_path']
      else
```

Reply:
> Wonder if we could combine `base_db_path` and `absolute_base_db_path` into a single config variable?
> 
> Did you try this?

It certainly occurred to me, but the current expectation is that `base_db_path` is relative to the application root, so didn't want to just change that behaviour without managed deprecation. I opted not to do that now given the lack of existing testing. It would reduce some complexity though.

Reply:
Oh, actually, it looks like when given a path starting `/`, `Rails.root.join` doesn't actually join it relative to `Rails.root`?!

```ruby
Rails.root.join('/foo').to_s
# => "/foo"

Rails.root.join('foo').to_s
# => "/home/vagrant/alaveteli/foo"
```

So basically this Just Works without this hunk of the commit. 🤦

Reply:
Oh nice - TIL